presto-main/src/main/java/com/facebook/presto/operator/scalar/ZipWithFunction.java

```diff
@@ -107,8 +132,11 @@ public static Block zipWith(Type leftElementType, Type rightElementType, Type ou
             catch (Throwable throwable) {
                 throw Throwables.propagate(throwable);
             }
-            writeNativeValue(outputElementType, resultBuilder, output);
+            writeNativeValue(outputElementType, blockBuilder, output);
```

Comment:
Make sure `pageBuilder` is in a consistent state before you throw. i.e. You probably need some code before `throw Throwables.propagate(throwable)`.

In addition, since you are refactoring this code, remove usage of the deprecated function `propagate`.